Lombiq.NodeJs.Extensions.Tests/MutexReaderWriterTest.cs

```diff
@@ -0,0 +1,83 @@
+using Lombiq.NodeJs.Extensions.CustomExecTasks;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Lombiq.NodeJs.Extensions.Tests;
+
+public class MutexReaderWriterTest
+{
+    private const string MutexName = nameof(MutexReaderWriterTest);
+    private const int ReaderWriterCount = 10;
+    private const int ReaderExecutionTimeMs = 1000;
+    private const int WriterExecutionTimeMs = 3000;
+    private const int MaxWaitTimeMs = 5000;
+
+    private readonly ITestOutputHelper _testOutputHelper;
+
+    public MutexReaderWriterTest(ITestOutputHelper testOutputHelper) => _testOutputHelper = testOutputHelper;
+
+    [Fact]
+    public void SyncWriterAndReaders()
+    {
+        var timeout = TimeSpan.FromMilliseconds(MaxWaitTimeMs);
+        var tasks = new List<Task>();
+        for (int i = 1; i <= ReaderWriterCount; i++)
+        {
+            var action = RandomNumberGenerator.GetInt32(0, 2) < 1 ? Reader(i, timeout) : Writer(i, timeout);
+            tasks.Add(Task.Run(action));
+        }
+
+        Task.WaitAll(tasks.ToArray());
+    }
```

Comment:
Please use meaningful names (`i, Reader, Writer`).
```suggestion
    public void SyncWriterAndReaders()
    {
        var timeout = TimeSpan.FromMilliseconds(MaxWaitTimeMs);
        var tasks = new List<Task>();

        for (var actionIndex = 1; actionIndex <= ReaderWriterCount; actionIndex++)
        {
            var action = RandomNumberGenerator.GetInt32(0, 2) < 1
                ? CreateReaderAction(actionIndex, timeout)
                : CreateWriterAction(actionIndex, timeout);
            tasks.Add(Task.Run(action));
        }

        Task.WaitAll(tasks.ToArray());
    }
```